src/core/Akka/Dispatch/ActorTaskScheduler.cs

```diff
@@ -57,8 +57,8 @@ protected override void QueueTask(Task task)
         }
 
         private void ScheduleTask(Task task)
-        {
-            _actorCell.Self.Tell(new ActorTaskSchedulerMessage(this, task), ActorRefs.NoSender);
+        {            
+            _actorCell.Self.Tell(new ActorTaskSchedulerMessage(this, task, _actorCell.CurrentMessage), ActorRefs.NoSender);
```

Comment:
Hmm wait, I think this approach doesn't work if the task is scheduled from outside of the actor's context.